### PR TITLE
fix(core): Block Public API related REST calls when Public API is not enabled

### PR DESCRIPTION
**Pull request by netroy**
When Public API is disabled via config, or unavailable because of licensing, the related rest endpoints to manage API keys should also be disabled.  

## Related tickets and issues
https://linear.app/n8n/issue/SEC-5

## Review / Merge checklist
- [x] PR title and summary are descriptive
- [x] Tests included

### CODE REVIEW

File: packages/cli/test/integration/me.api.test.ts
Comment (despairblue) [inline packages/cli/test/integration/me.api.test.ts:1]: Is there a reason why `DELETE` is not tested?

File: packages/cli/src/services/frontend.service.ts [hunk 1]
```diff
@@ -143,7 +144,7 @@ export class FrontendService {
 				},
 			},
 			publicApi: {
-				enabled: !config.get('publicApi.disabled') && !this.license.isAPIDisabled(),
+				enabled: isApiEnabled(),
```
[the remark this answers is not in the document]
Comment (netroy) [reply packages/cli/src/services/frontend.service.ts:147]: it gets mapped to `api.enabled` in `ISettingsState` in the frontend. 
the inconsistent naming between frontend and backend has confused me before as well.

File: packages/cli/test/integration/me.api.test.ts
Comment (netroy) [reply packages/cli/test/integration/me.api.test.ts:1]: updated

**Review by despairblue — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.44.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.44.0)

**Pull request merged**